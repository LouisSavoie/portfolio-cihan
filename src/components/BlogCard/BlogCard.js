import React from 'react'
import Card from 'react-bootstrap/Card'

const BlogCard = () => {
  return (
    <a data-testid="blog-card" href="">
      <Card style={{ width: '18rem' }}>
        <Card.Img variant="top" src="holder.js/100px180" />
        <Card.Body>
          <Card.Title>Title</Card.Title>
          <Card.Text>Text</Card.Text>
          <Card.Text>Date</Card.Text>
        </Card.Body>
      </Card>
    </a>
  )
}

export default BlogCard