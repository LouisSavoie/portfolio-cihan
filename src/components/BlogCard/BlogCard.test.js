import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import BlogCard from './BlogCard'

const blogItem = {
  title: 'Failed to fetch blog posts :(',
  link: 'https://medium.com/@thelousavoie',
  thumbnail: './img/logo512.png',
  content: 'The RSS feed for thelousavoie is temporarily unavailable. Please check back again later! Sorry, for the inconvenience.',
  pubDate: '0000-00-00 00:00:00'
}

afterEach(() => {
  cleanup()
})

test('should render BlogCard', () => {
  render(<BlogCard blogItem={blogItem} />)
  const BlogCardElement = screen.getByTestId('blog-card')
  expect(BlogCardElement).toBeInTheDocument()
})

test('matches snapshot', () => {
  const tree = renderer.create(<BlogCard blogItem={blogItem} />).toJSON()
  expect(tree).toMatchSnapshot()
})
