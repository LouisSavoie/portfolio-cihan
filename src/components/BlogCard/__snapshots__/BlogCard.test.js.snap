// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card"
  style={
    Object {
      "width": "24rem",
    }
  }
>
  <a
    data-testid="blog-card"
    href="https://medium.com/@thelousavoie"
  >
    <img
      alt="Failed to fetch blog posts :("
      className="card-img-top card-img-top-center-cropped"
      src="./img/logo512.png"
    />
    <div
      className="card-body"
    >
      <h3
        className="card-title"
      >
        Failed to fetch blog posts :(
      </h3>
      <p
        className="card-text"
      >
        The RSS feed for thelousavoie is temporarily unavailable. Please check back again later! Sorry, for the inconvenience....
      </p>
      <h4
        className="card-title"
      >
        Published: 00/00/0000
      </h4>
    </div>
  </a>
</div>
`;
